kumascript/macros/JSRef.ejs

```diff
@@ -116,24 +116,45 @@ result['iObject'] = {
 
 var pageList;
 var isObj;
-var includeme;
+
+function isPrototypeMemberPage(page) {
+    return [
+        "javascript-instance-accessor-property",
+        "javascript-instance-data-property",
+        "javascript-instance-method",
+    ].includes(page.pageType);
+}
+
+function isPropertyPage(page) {
+    return [
+        "javascript-static-accessor-property",
+        "javascript-static-data-property",
+        "javascript-instance-accessor-property",
+        "javascript-instance-data-property",
+    ].includes(page.pageType);
+}
+
+function isMethodPage(page) {
+    return [
+        "javascript-static-method",
+        "javascript-instance-method",
+    ].includes(page.pageType);
+}
 
 for (var object in source) {
     pageList = source[object];
     if (object == "iObject") {
         isObj = true;
     }
-    for (aPage in pageList) {
-        if (isObj) {
-            includeme = containsTag(pageList[aPage], "prototype");
-        } else {
-            includeme = true;
-        }
-        if (containsTag(pageList[aPage], 'Property') && includeme) {
-            result[object].properties.push(pageList[aPage]);
+    for (const page of pageList) {
+        if (isObj && !isPrototypeMemberPage(page)) {
+            continue;
```

Comment:
Nit: We don't really need the `isObj` variable, and could filter instead. This avoids the negation and makes the code easier to understand.

```js
for (var object in source) {
    pageList = source[object];

    if (object == "iObject") {
        // For objects, we're only interested in prototype members.
        pageList = pageList.filter(page => isPrototypeMemberPage(page));
    }

    for (const page of pageList) {
        if (isPropertyPage(page)) {
            result[object].properties.push(page);
          }
        }
        if (isMethodPage(page)) {
            result[object].methods.push(page);
        }

Reply:
Yes, much better!